fs/generic.c

```diff
@@ -287,3 +287,22 @@ int generic_rmdirat(struct fd *at, const char *path_raw) {
     mount_release(mount);
     return err;
 }
+
+int generic_seek(struct fd *fd, off_t_ off, int whence) {
+    off_t_ new_off = fd->offset;
+    if (whence == LSEEK_SET) {
+        fd->offset = off;
+    } else if (whence == LSEEK_CUR) {
+        if (new_off += off < 0)
```

Comment:
I've been putting off fixing this because I think it's a little bit harder to fix than this. Something like "off < 0 && fd->offset > old_off"